File: StudyHub.Frontend/src/exam/interfaces/models/ExamResult.ts
import type { Answer } from "./Answer";

export interface ExamResult {
  id: string,
  examId: number,
  studentId: string,
  studentName?: string,
  finishTime: Date,
  submissionTime?: Date,
  score?: number,
  answers: Answer[],
  cheatTimes: number
}

File: StudyHub.Backend/StudyHub.Backend.Domain/Entities/Exam/ExamResult.cs
namespace StudyHub.Backend.Domain.Entities.Exam
{
    public class ExamResult
    {
        public string Id { get; set; } = string.Empty;
        public int ExamId { get; set; }
        public List<ExamAnswer> Answers { get; set; } = new List<ExamAnswer>();
        public Guid StudentId { get; set; }
        public string StudentName { get; set; } = string.Empty;
        public int CheatTimes { get; set; } = 0;
        public DateTime FinishTime { get; set; }
        public DateTime? SubmissionTime { get; set; }
        public int TotalQuestions { get; set; }
        public decimal Score { get; set; }
    }
}
